fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeMetadata.java

```diff
@@ -70,6 +108,108 @@ public List<String> listPartitionNames(String databaseName, String tableName, lo
+    }
+
+    private void triggerDeltaLakePlanFilesIfNeeded(PredicateSearchKey key, Table table, ScalarOperator operator,
+                                                   Tracers tracers) {
+        if (!scannedTables.contains(key)) {
+            tracers = tracers == null ? Tracers.get() : tracers;
+            try (Timer ignored = Tracers.watchScope(tracers, EXTERNAL, "DELTA_LAKE.processSplit." + key)) {
+                collectDeltaLakePlanFiles(key, table, operator);
+            }
+        }
+    }
+
+    private void collectDeltaLakePlanFiles(PredicateSearchKey key, Table table, ScalarOperator operator) {
+        DeltaLakeTable deltaLakeTable = (DeltaLakeTable) table;
+        Metadata metadata = deltaLakeTable.getDeltaMetadata();
+        Engine engine = deltaLakeTable.getDeltaEngine();
+        SnapshotImpl snapshot = (SnapshotImpl) deltaLakeTable.getDeltaSnapshot();
+
+        StructType schema = metadata.getSchema();
+        Set<String> partitionColumns = metadata.getPartitionColNames();
+
+        List<ScalarOperator> scalarOperators = Utils.extractConjuncts(operator);
+        ScalarOperationToDeltaLakeExpr.DeltaLakeContext deltaLakeContext =
+                new ScalarOperationToDeltaLakeExpr.DeltaLakeContext(schema, partitionColumns);
+        Predicate deltaLakePredicate = new ScalarOperationToDeltaLakeExpr().convert(scalarOperators, deltaLakeContext);
+
+        ScanBuilder scanBuilder = snapshot.getScanBuilder(engine);
+        Scan scan = scanBuilder.withFilter(engine, deltaLakePredicate).build();
+
+        List<Row> files = Lists.newArrayList();
+
+        try (CloseableIterator<FilteredColumnarBatch> scanFilesAsBatches = scan.getScanFiles(engine)) {
+            while (scanFilesAsBatches.hasNext()) {
+                FilteredColumnarBatch scanFileBatch = scanFilesAsBatches.next();
+                try (CloseableIterator<Row> scanFileRows = scanFileBatch.getRows()) {
+                    while (scanFileRows.hasNext()) {
+                        Row scanFileRow = scanFileRows.next();
+                        DeletionVectorDescriptor dv = InternalScanFileUtils.getDeletionVectorDescriptorFromRow(scanFileRow);
+                        if (dv != null) {
+                            ErrorReport.reportValidateException(ErrorCode.ERR_BAD_TABLE_ERROR, ErrorType.UNSUPPORTED,
+                                    "Delta table feature [deletion vectors] is not supported");
+                        }
+                        files.add(scanFileRow);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            LOG.error("Failed to get delta lake scan files", e);
+            throw new StarRocksConnectorException("Failed to get delta lake scan files", e);
+        }
+
+        splitTasks.put(key, files);
+        scannedTables.add(key);
+    }
+
+    @Override
+    public boolean prepareMetadata(MetaPreparationItem item, Tracers tracers, ConnectContext connectContext) {
```

Comment:
why do you implement these interfaces in this patch? if you want to support the `prepareMetadata`, you need to implement `supportPreCollectMetadata` in the DeltalakeTable.

Reply:
Yes, you are right, i will remove it now.